spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -366,9 +365,9 @@ public Scan buildCopyOnWriteScan() {
 
     Schema expectedSchema = schemaWithMetadataColumns();
 
-    TableScan scan =
+    BatchScan scan =
```

Comment:
Had to change this so that `SparkCopyOnWriteScan` can use the same task planning.